pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -162,3 +135,90 @@ func (r *PullSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Named(controllers.PullSecretControllerName).
 		Complete(r)
 }
+
+// fixAndUpdateGlobalPullSecret checks the state of the pull secrets, in case of missing or broken ARO pull secret
+// it replaces it with working one from controller Secret
+func (r *PullSecretReconciler) fixAndUpdateGlobalPullSecret(ctx context.Context, operatorSecret, userSecret *corev1.Secret) (secret *corev1.Secret, err error) {
+	if operatorSecret == nil {
+		return nil, errors.New("nil operator secret, cannot verify userData integrity")
+	}
+
+	create := userSecret == nil
+	remove := false
+
+	// userSecret can happen to have broken type, which means it have to be recreated
+	// with proper type
+	if userSecret != nil && (userSecret.Type != corev1.SecretTypeDockerConfigJson || userSecret.Data == nil) {
+		create = true
+		remove = true
+	}
```

Comment:
I got a little confused here. I didn't understand at first how both `create` and `remove` could be true.

`remove` is true if and only if "`create` is true because the secret existed but it was bad."

Maybe a better name for `remove` would be `replace`.

Additionally, we could make the logic easier to read like so:

```suggestion
	var create, replace bool
	if userSecret == nil {
		create = true
	} else if userSecret.Type != corev1.SecretTypeDockerConfigJson || userSecret.Data == nil {
		create, replace = true, true
	}
```

Reply:
I have actually changed the whole function to make it more maintainable. The thought is to make the flow easier to read, but sacrifice one deletion call to the API. 
The delete happens only when there is a change, so there is no significant performance penalty.